src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -303,6 +308,82 @@ def optimize(self, verbose=False):
         if verbose:
             print(f"optimization completed.")
         return
+    
+    def evaluate_circuit(
+        self, params: Union[List[float], Dict[str, List[float]]]
```

Comment:
Question: Should you be able to pass a QAOAVariationalBaseParams object too?

Reply:
I have updated that, but the problem that I see is that you could pass an object with different properties than the QAOA object compiled. For example:
```
q = QAOA()
q.compile(problem)

params_obj = (create a params obj with p>1)
q.evaluate_circuit(params_obj)
```

Reply:
I guess we can just raise an error for this case.

Reply:
I have added some checks for this